File: ios/Classes/LeancloudArgsConverter.m
//
//  LeancloudArgsConverter.m
//  leancloud_flutter_plugin
//
//  Created by Luna on 2019/1/6.
//

#import "LeancloudArgsConverter.h"

@implementation LeancloudArgsConverter

+ (NSDictionary*) getAVQueryJsonObject:(FlutterMethodCall*)call result:(FlutterResult)result {
    id arg = call.arguments[@"avQuery"];
    if (arg == nil) {
        result([FlutterError errorWithCode:@"missing-arg"
                                   message:@"Arg '%@' can't be null, set empty value. PLEASE FIX IT!"
                                   details:nil]);
        return nil;
    } else {
        return [LeancloudArgsConverter stringToJson:[NSString stringWithFormat:@"%@", arg]];
    }
}

+ (NSString*) getStringValue:(FlutterMethodCall*)call result:(FlutterResult)result key:(NSString*)key {
    id arg = call.arguments[key];
    if (arg == nil) {
        result([FlutterError errorWithCode:@"missing-arg"
                                   message:@"Arg '%@' can't be null, set empty value. PLEASE FIX IT!"
                                   details:nil]);
    } else {
        if ([arg isKindOfClass:[NSString class]]) {
            return [NSString stringWithFormat:@"%@", arg];
        } else {
            result([FlutterError errorWithCode:@"arg-type-error"
                                       message:@"Arg '%@' should be String. PLEASE FIX IT!"
                                       details:nil]);
        }
    }
    return @"";
}

+ (int) getIntValue:(FlutterMethodCall*)call result:(FlutterResult)result key:(NSString*)key {
    id arg = call.arguments[key];
    if (arg == nil) {
        result([FlutterError errorWithCode:@"missing-arg"
                                   message:@"Arg '%@' can't be null, set empty value. PLEASE FIX IT!"
                                   details:nil]);
    } else {
        if ([arg isKindOfClass:[NSNumber class]]) {
            return [arg intValue];
        } else {
            result([FlutterError errorWithCode:@"arg-type-error"
                                       message:@"Arg '%@' should be int. PLEASE FIX IT!"
                                       details:nil]);
        }
    }
    return 0;
}

+ (NSDictionary*) stringToJson:(NSString*) jsonString {
    NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
    NSError *err;
    NSDictionary *jsonObject = [NSJSONSerialization JSONObjectWithData:jsonData
                                                               options:NSJSONReadingMutableContainers
                                                                 error:&err];
    if(err)
    {
        NSLog(@"Json ERROR: %@",err);
        return nil;
    }
    return jsonObject;
}

@end

File: android/src/main/java/com/lunagao/leancloudflutterplugin/LeancloudArgsConverter.java
package com.lunagao.leancloudflutterplugin;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;

class LeancloudArgsConverter {

    static JSONObject getAVQueryJsonObject(MethodCall call, MethodChannel.Result result) {
        String key = "avQuery";
        Object arg = call.argument("avQuery");
        if (arg == null) {
            result.error("missing-arg", "Arg '" + key + "' can't be null, set empty value. PLEASE FIX IT!", null);
            return null;
        } else {
            return JSON.parseObject(arg.toString());
        }
    }

    static String getStringValue(MethodCall call, MethodChannel.Result result, String key) {
        Object arg = call.argument(key);
        if (arg == null) {
            result.error("missing-arg", "Arg '" + key + "' can't be null, set empty value. PLEASE FIX IT!", null);
            return "";
        } else {
            return arg.toString();
        }
    }

    static int getIntValue(MethodCall call, MethodChannel.Result result, String key) {
        Object arg = call.argument(key);
        if (arg == null) {
            result.error("missing-arg", "Arg '" + key + "' can't be null, set 0 value. PLEASE FIX IT!", null);
            return 0;
        } else {
            return (int)arg;
        }
    }
}
